leakcanary-android-core/src/main/java/leakcanary/internal/activity/LeakActivity.kt

```diff
@@ -60,6 +60,18 @@ internal class LeakActivity : NavigatingActivity() {
     leaksButton.setOnClickListener { resetTo(LeaksScreen()) }
     heapDumpsButton.setOnClickListener { resetTo(HeapDumpsScreen()) }
     aboutButton.setOnClickListener { resetTo(AboutScreen()) }
+
+    handleViewHprof(intent)
+  }
+
+  private fun handleViewHprof(intent: Intent?) {
+    if (intent?.action != Intent.ACTION_VIEW) return
+    val uri = intent.data ?: return
+    if (uri.lastPathSegment?.endsWith(".hprof") != true) return
+    resetTo(HeapDumpsScreen())
```

Comment:
This is ok but based on the video there's a little amount of time where one might think they have to click on "import hprof" on that new screen, when in fact the hprof was already being imported. So now I'm wondering if we should add a virtual row for "pending analysis" on that screen (for this new feature but also any time we import).

Reply:
That would be a nice addition indeed.
I'll try to look into that, but it seems quite more involving (modifying HeapAnalysisTable.Projection I guess).

Reply:
I'm not sure we'd want to update the DB as this is very much an "in progress" thing that's only true while the program is running, so we'd risk having out of date info after a restart.

We could probably file this as a follow up... but I'm thinking maybe an internal thing that keeps track of "are we running an analysis right now" and a way to subscribe to that.